File: simple_kv/src/main/java/com/jackzhao/simple_kv/IBaseKv.java
package com.jackzhao.simple_kv;

import android.content.Context;

public interface IBaseKv {

    default String getFileName() {
        return this.getClass().getSimpleName();
    }

    default Object get(Context context, Object type) {
        return null;
    }

    default void set(Context context, Object v) {
    }

    default int increase(Context context) {
        return 0;
    }

    default void reset(Context context) {
    }

    default IBaseKv getReal() {
        IBaseKvProxy proxy = new IBaseKvProxy();
        proxy.newProxyInstance(this);
        return (IBaseKv) proxy.getProxyInstance();
    }
}

File: simple_kv/src/main/java/com/jackzhao/simple_kv/IBaseKv.kt
package com.jackzhao.simple_kv

interface IBaseKv {
    val fileName: String
        get() = this.javaClass.simpleName

    fun get(): Any? {
        return proxy.get()
    }

    fun getBoolean(): Boolean {
        return proxy.getBoolean()
    }

    fun getInt(): Int {
        return proxy.getInt()
    }

    fun getLong(): Long {
        return proxy.getLong()
    }

    fun getString(): String {
        return proxy.getString()
    }

    fun getHashSet(): HashSet<Any?> {
        return proxy.getHashSet()
    }

    fun set(v: Any?) {
        proxy.set(v)
    }

    fun increase(): Int {
        return proxy.increase()
    }

    fun reset() {
        proxy.reset()
    }

    val proxy: IBaseKv
        get() {
            val proxy = IBaseKvProxy()
            proxy.newProxyInstance(this)
            return proxy.proxyInstance as IBaseKv
        }
}